code/modules/mining/equipment/marker_beacons.dm

```diff
@@ -0,0 +1,126 @@
+	amount = 10
+
+/obj/item/stack/marker_beacon/thirty //and they're bought in stacks of 1, 10, or 30
+	amount = 30
+
+/obj/item/stack/marker_beacon/Initialize(mapload)
+	. = ..()
+	update_icon()
+
+/obj/item/stack/marker_beacon/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Use in-hand to place a [singular_name].</span>")
+	to_chat(user, "<span class='notice'>Alt-click to select a color. Current color is [picked_color].</span>")
+
+/obj/item/stack/marker_beacon/update_icon()
+	icon_state = "[initial(icon_state)][lowertext(picked_color)]"
+
+/obj/item/stack/marker_beacon/attack_self(mob/user)
+	if(!isturf(user.loc))
+		to_chat(user, "<span class='warning'>You need more space to place a [singular_name] here.</span>")
+		return
+	if(locate(/obj/structure/marker_beacon) in user.loc)
+		to_chat(user, "<span class='warning'>There is already a [singular_name] here.</span>")
+		return
+	if(use(1))
+		to_chat(user, "<span class='notice'>You activate and anchor [amount ? "a":"the"] [singular_name] in place.</span>")
+		playsound(user, 'sound/machines/click.ogg', 50, 1)
+		var/obj/structure/marker_beacon/M = new /obj/structure/marker_beacon(user.loc, picked_color)
+		transfer_fingerprints_to(M)
+
+/obj/item/stack/marker_beacon/AltClick(mob/user)
+	if(user.incapacitated())
+		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
+		return
+	if(!in_range(src, user))
+		return
+	var/input_color = input(user, "Choose a color.", "Beacon Color") as null|anything in GLOB.marker_beacon_colors
+	if(input_color)
+		picked_color = input_color
+		update_icon()
+
+/obj/structure/marker_beacon
+	name = "marker beacon"
+	desc = "A Canary-brand path illumination device. It is anchored in place and glowing steadily."
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "marker"
+	layer = BELOW_OPEN_DOOR_LAYER
+	anchored = TRUE
+	light_range = 2
+	light_power = 3
+	var/remove_speed = 15
+	var/picked_color
+
+/obj/structure/marker_beacon/Initialize(mapload, set_color)
+	. = ..()
+	picked_color = set_color
+	update_icon()
+
+/obj/structure/marker_beacon/examine(mob/user)
+	..()
+	to_chat(user, "<span class='notice'>Alt-click to select a color. Current color is [picked_color].</span>")
+
+/obj/structure/marker_beacon/update_icon()
+	while(!picked_color || !GLOB.marker_beacon_colors[picked_color])
+		picked_color = pick(GLOB.marker_beacon_colors)
+	icon_state = "[initial(icon_state)][lowertext(picked_color)]-on"
+	set_light(light_range, light_power, GLOB.marker_beacon_colors[picked_color])
+
+/obj/structure/marker_beacon/attack_hand(mob/living/user)
+	to_chat(user, "<span class='notice'>You start picking [src] up...</span>")
+	if(do_after(user, remove_speed, target = src))
+		var/obj/item/stack/marker_beacon/M = new(loc)
+		transfer_fingerprints_to(M)
+		if(user.put_in_hands(M, TRUE)) //delete the beacon if it fails
+			playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
+			qdel(src) //otherwise delete us
```

Comment:
unconditional? it's not unconditional, it only happens if you picked the beacon up, this just deletes the structure that would otherwise remain.